### ISSUE TITLE
Unable to use history argument while a backup is in progress

### ISSUE THREAD

**Issue by fatman00** [cskoglun/ciscodnacbackupctl#8]
ciscodnacbackupctl version 0.2.5
Ubuntu 20.04
Python 3.8.5
DNAC version 2.1.2.7

When a backup is running I am not able to list the history using the tool. The data is available using the DNAC GUI.
```
user@dnabackup:~$ ciscodnacbackupctl daemon start --keep 5
Starting ciscodnacbackupctl ... OK
user@dnabackup:~$ ciscodnacbackupctl create --name nu02
Creating backup with id: 26c58667-800f-4be3-aca9-710cf1847d33
user@dnabackup:~$ ciscodnacbackupctl create --name nu03
Error: Can not perform BACKUP due to ongoing workflow: BACKUP 26C58667-800F-4BE3-ACA9-710CF1847D33. Please try again at a later time.
user@dnabackup:~$ ciscodnacbackupctl progress
BACKUP_ID                             BACKUP_SIZE    DESCRIPTION    ID                                    OPERATION    PROGRESS_IN_PERCENTAGE    START_TIMESTAMP      STATUS
26c58667-800f-4be3-aca9-710cf1847d33  0B             nu02           26c58667-800f-4be3-aca9-710cf1847d33  BACKUP       10%                       2021-06-24 07:03:16  PENDING
user@dnabackup:~$ ciscodnacbackupctl history
user@dnabackup:~$ ciscodnacbackupctl history
user@dnabackup:~$ ciscodnacbackupctl progress
BACKUP_ID                             BACKUP_SIZE    DESCRIPTION    ID                                    OPERATION    PROGRESS_IN_PERCENTAGE    START_TIMESTAMP      STATUS
26c58667-800f-4be3-aca9-710cf1847d33  198M           nu02           26c58667-800f-4be3-aca9-710cf1847d33  BACKUP       50%                       2021-06-24 07:03:16  PENDING
user@dnabackup:~$ ciscodnacbackupctl list
BACKUP_ID                             BACKUP_SIZE    COMPATIBLE    DESCRIPTION    END_TIMESTAMP        START_TIMESTAMP      STATUS
ebc337d0-53a2-4e04-acff-4fa0b7a41434  5G             TRUE          daily          2021-06-20 22:03:02  2021-06-20 22:00:31  SUCCESS
3c666096-575c-4af1-b2a5-69e453932fc2  5G             TRUE          daily          2021-06-23 22:03:07  2021-06-23 22:00:35  SUCCESS
1faae89d-7c69-43f7-80a6-a49488804efd  5G             TRUE          nu01           2021-06-24 07:01:59  2021-06-24 06:59:23  SUCCESS
user@dnabackup:~$ ciscodnacbackupctl progress
BACKUP_ID                             BACKUP_SIZE    DESCRIPTION    ID                                    OPERATION    PROGRESS_IN_PERCENTAGE    START_TIMESTAMP      STATUS
26c58667-800f-4be3-aca9-710cf1847d33  319M           nu02           26c58667-800f-4be3-aca9-710cf1847d33  BACKUP       60%                       2021-06-24 07:03:16  PENDING
user@dnabackup:~$ ciscodnacbackupctl progress
user@dnabackup:~$ ciscodnacbackupctl list
BACKUP_ID                             BACKUP_SIZE    COMPATIBLE    DESCRIPTION    END_TIMESTAMP        START_TIMESTAMP      STATUS
ebc337d0-53a2-4e04-acff-4fa0b7a41434  5G             TRUE          daily          2021-06-20 22:03:02  2021-06-20 22:00:31  SUCCESS
3c666096-575c-4af1-b2a5-69e453932fc2  5G             TRUE          daily          2021-06-23 22:03:07  2021-06-23 22:00:35  SUCCESS
1faae89d-7c69-43f7-80a6-a49488804efd  5G             FALSE         nu01           2021-06-24 07:01:59  2021-06-24 06:59:23  SUCCESS
26c58667-800f-4be3-aca9-710cf1847d33  5G             TRUE          nu02           2021-06-24 07:07:01  2021-06-24 07:03:28  SUCCESS
user@dnabackup:~$ ciscodnacbackupctl history
BACKUP_ID                             BACKUP_SIZE    DESCRIPTION    ID                                    OPERATION    PROGRESS_IN_PERCENTAGE    START_TIMESTAMP      STATUS
db8e81b8-de2a-4313-a66f-92ffd3039bfe  5G             tester         db8e81b8-de2a-4313-a66f-92ffd3039bfe  BACKUP       100%                      2021-06-17 08:58:31  SUCCESS
b28e10bc-bd44-4954-accb-c3de10f0c1b4  5G             test02         b28e10bc-bd44-4954-accb-c3de10f0c1b4  BACKUP       100%                      2021-06-17 09:13:32  SUCCESS
df64a1db-ce90-40b0-91af-788674b49d7a  5G             nu01           df64a1db-ce90-40b0-91af-788674b49d7a  BACKUP       100%                      2021-06-17 12:50:20  SUCCESS
4c115da4-6fd8-4944-a2ed-1be0ae37d5c6  5G             nu02           4c115da4-6fd8-4944-a2ed-1be0ae37d5c6  BACKUP       100%                      2021-06-17 12:54:01  SUCCESS
c1bc61a8-a4e5-40c0-b710-5a4fa09eca1f  5G             nu03           c1bc61a8-a4e5-40c0-b710-5a4fa09eca1f  BACKUP       100%                      2021-06-17 12:58:57  SUCCESS
4e20015d-2a1a-4814-8a91-81d9ce8b148e  5G             nu04           4e20015d-2a1a-4814-8a91-81d9ce8b148e  BACKUP       100%                      2021-06-17 13:05:39  SUCCESS
3a8e7f1a-d85a-4982-9acb-20dcadbee238  5G             nu05           3a8e7f1a-d85a-4982-9acb-20dcadbee238  BACKUP       100%                      2021-06-17 13:12:36  SUCCESS
faf5afb6-4c3b-42cb-9a7c-2db9645034bb  5G             nu05           faf5afb6-4c3b-42cb-9a7c-2db9645034bb  BACKUP       100%                      2021-06-17 13:36:58  SUCCESS
ac78ac99-ed8d-4e0e-a40d-180afceefed2  5G             nu06           ac78ac99-ed8d-4e0e-a40d-180afceefed2  BACKUP       100%                      2021-06-17 13:56:07  SUCCESS
52d8c2ad-24e4-4b6a-8027-0a53be80f315  5G             daily          52d8c2ad-24e4-4b6a-8027-0a53be80f315  BACKUP       100%                      2021-06-22 22:00:26  SUCCESS
3c666096-575c-4af1-b2a5-69e453932fc2  5G             daily          3c666096-575c-4af1-b2a5-69e453932fc2  BACKUP       100%                      2021-06-23 22:00:24  SUCCESS
1faae89d-7c69-43f7-80a6-a49488804efd  5G             nu01           1faae89d-7c69-43f7-80a6-a49488804efd  BACKUP       100%                      2021-06-24 06:59:12  SUCCESS
26c58667-800f-4be3-aca9-710cf1847d33  5G             nu02           26c58667-800f-4be3-aca9-710cf1847d33  BACKUP       100%                      2021-06-24 07:03:16  SUCCESS
user@dnabackup:~$
```

**Comment by robertcsapo** [comment]
```26c58667-800f-4be3-aca9-710cf1847d33  319M           nu02           26c58667-800f-4be3-aca9-710cf1847d33  BACKUP       60%                       2021-06-24 07:03:16  PENDING```

Needs to be completed, until it's shown in ```history```could you verify that the backup was completed?

**Comment by fatman00** [comment]
Hi,
The issue is not with the history command, but the list command.
When a backup is in progress the `ciscodnacbackupctl list` command does not give any output.

The backups completed successful.

**Comment by robertcsapo** [comment]
@fatman00 for me it's the ```history``` commands that return nothing during backup.

```
% ciscodnacbackupctl history
BACKUP_ID                             BACKUP_SIZE    DESCRIPTION    ID                                    OPERATION    PROGRESS_IN_PERCENTAGE    START_TIMESTAMP      STATUS
34529229-7f0d-4102-9203-1a71044c3b50  19G            dnac-weekly    34529229-7f0d-4102-9203-1a71044c3b50  BACKUP       100%                      2021-07-04 01:00:22  SUCCESS
d50e58e2-6511-4c29-92d9-1b274382f28f  20G            dnac-weekly    d50e58e2-6511-4c29-92d9-1b274382f28f  BACKUP       100%                      2021-07-11 01:00:24  SUCCESS
f551d912-2e9c-43e5-a97b-145bba96bce7  20G            dnac-weekly    f551d912-2e9c-43e5-a97b-145bba96bce7  BACKUP       100%                      2021-07-18 01:00:25  SUCCESS
b29b432a-8ee6-47d4-b721-52ace07bba41  20G            dnac-weekly    b29b432a-8ee6-47d4-b721-52ace07bba41  BACKUP       100%                      2021-07-25 01:00:23  SUCCESS
e3984c31-a084-41fd-a215-204be6c2e989  20G            dnac-weekly    e3984c31-a084-41fd-a215-204be6c2e989  BACKUP       100%                      2021-08-01 01:00:30  SUCCESS
```

```
% ciscodnacbackupctl progress
BACKUP_ID                             BACKUP_SIZE    DESCRIPTION    ID                                    OPERATION    PROGRESS_IN_PERCENTAGE    START_TIMESTAMP      STATUS
cd3183bd-2c1f-49a5-a9b3-ff264d2e8d62  408M           nu02           cd3183bd-2c1f-49a5-a9b3-ff264d2e8d62  BACKUP       50%                       2021-08-04 08:53:28  PENDING

% ciscodnacbackupctl history 
%

**Comment by robertcsapo** [comment]
@fatman00 fixed in 0.2.6 version
```pip install ciscodnacbackupctl==0.2.6```

thanks for reporting this! 👍 😄
